### PR TITLE
chore: add luacheck workflow for PRs

### PR DESCRIPTION
**Pull request by bahrmichael**
Introduces static code analysis for lua files. Should prevent problems like https://github.com/bahrmichael/factorio-tycoon/commit/4579b5de58fa6cb04374a20d2a50cc914a132813.

### CODE REVIEW

**Comment by winex** [comment]
yeah, that's cool. i tried same thing after my `pr-1`, but didn't return to it.
i used Nexela's `.luacheckrc`, you may check my `winex:test` branch https://github.com/bahrmichael/factorio-tycoon/compare/0.4.1...winex:factorio-tycoon:test if it has anything useful

badge looks cool, btw :)

File: control_old.lua
Comment (winex) [inline control_old.lua:1]: sometimes i look into and `grep` this file, but it is becoming too old now and we can always search in history...

**Comment by winex** [comment]
```
prototypes/entities/hidden-light.lua:3:23: accessing undefined field deepcopy of global table
```
- `table.deepcopy()` should be added in someway as correct (but it's only in `data` stage, iirc)
- `table_size()` func as well, provided by Factorio

**Comment by bahrmichael** [comment]
> yeah, that's cool. i tried same thing after my `pr-1`, but didn't return to it. i used Nexela's `.luacheckrc`, you may check my `winex:test` branch [0.4.1...winex:factorio-tycoon:test](https://github.com/bahrmichael/factorio-tycoon/compare/0.4.1...winex:factorio-tycoon:test) if it has anything useful
> 
> badge looks cool, btw :)

Thanks! I took a look, but prefer to stick with an actions and .rc file for now. I'll make the luacheck more strict over time, but want to wait until your big changes are in.

**Comment by bahrmichael** [comment]
> https://github.com/Nexela/Factorio-luacheckrc/tree/master/.luacheckrc#L338-L349

Nice! I added that in https://github.com/bahrmichael/factorio-tycoon/commit/fc973f06151de9d1b805fbc775f523d07cb975ab

**Comment by bahrmichael** [comment]
Let's add that when we use it :) checks were green so i expected there to be no current issues

**Pull request merged**